ext/pdo_oci/oci_statement.c

```diff
@@ -602,9 +602,8 @@ static int oci_stmt_describe(pdo_stmt_t *stmt, int colno) /* {{{ */
 			} else {
 				S->cols[colno].datalen = (ub4) col->maxlen;
 			}
-			if (dtype == SQLT_BIN) {
-				S->cols[colno].datalen *= 3;
-			}
+
+			S->cols[colno].datalen *= S->H->max_char_width;
```

Comment:
Rationale from the developer who looked at it is:
"Given raw data(SQLT_BIN) is defined as SQLT_CHR (See line #609), raw is returned as strings. So 1 byte raw is represented as two bytes of hex digits.
For example, a 1-byte raw data 'FF' is converted to two 'F' characters. So we need to multiply by 2 (not 3) to accommodate the translation of raw to hex digits. `max_char_width` should be used for character strings.

Apart from that, other changes all look good including the testcase. The test would normally be skipped in the test suite as a UTF-8 DSN is required.

But we can see the skip reason in the test result and run it manually if required."